### PR TITLE
Fixes for self-installed MKL from intel oneAPI

### PR DESCRIPTION
**Pull request by j-fu**
* introduces Pardiso.mkl_is_available() for use in runtests.jl
* test load mkl in __init__ (like pardiso)
* some link updates, hint at LD_LIBRARY_PATH

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,20 +18,21 @@ library.
...
   for a thorough guide how to set this variable correctly, typically done by
-  executing something like `source /opt/intel/mkl/bin/mklvars.sh intel64` or
+  executing something like `source /opt/intel/oneapi/setvars.sh intel64` or
   running `"C:\Program Files (x86)\IntelSWTools\compilers_and_libraries\windows\mkl\bin\mklvars.bat" intel64`
-* Run `Pkg.build("Pardiso")`
+* Run `Pkg.build("Pardiso", verbose=true)`
 * Run `Pardiso.show_build_log()` to see the build log for additional
   information.
-* Note that the `MKLROOT` environment variable must be set whenever using the library.
+* Note that the `MKLROOT` environment variable must be set, and `LD_LIBRARY_PATH` must contain `$MKLROOT/lib`
```
Comment (KristofferC) [inline README.md:34]: Is this true? I don't think I have had to explicitly set this before.

File: src/mkl_pardiso.jl [hunk 2]
```diff
@@ -13,7 +13,7 @@ mutable struct MKLPardisoSolver <: AbstractPardisoSolver
 end
 
 function MKLPardisoSolver()
-    if !MKL_jll.is_available()
+    if !( LOCAL_MKL_FOUND || MKL_jll.is_available())
```
Comment (KristofferC) [inline src/mkl_pardiso.jl]:
```suggestion
    if !(LOCAL_MKL_FOUND || MKL_jll.is_available())
```

File: src/Pardiso.jl [hunk 3]
```diff
@@ -128,6 +131,15 @@ function __init__()
         @warn "MKLROOT not set, MKL Pardiso solver will not be functional"
     end
 
+    if mkl_is_available() 
+        try
+            libmklpardiso = Libdl.dlopen(libmkl_rt[])
```
Comment (KristofferC) [inline src/Pardiso.jl:139]: Should this make `mkl_is_available()` return false after it has failed to load it?

File: src/mkl_pardiso.jl (hunk 2 above)
Comment (KristofferC) [inline src/mkl_pardiso.jl]: use `mkl_is_available()` here?

File: src/mkl_pardiso.jl [hunk 4]
```diff
@@ -13,7 +13,7 @@ mutable struct MKLPardisoSolver <: AbstractPardisoSolver
...
         error("MKL is not available no this platform")
```
Comment (KristofferC) [inline src/mkl_pardiso.jl]: Maybe this error message is a bit too specific now (talking about the platform) when there could be other reasons MKL is not available

File: README.md (hunk 1 above)
Comment (j-fu) [reply README.md:34]: The advise by intel is to run
```
source /opt/intel/oneapi/setvars.sh
```
This sets MKLROOT and adds the lib directory to the LD_LIBRARY_PATH . I accidently run with only MKLROOT set, and libmkl_rt was not found as a consequence.

File: src/mkl_pardiso.jl (hunk 2 above)
Comment (j-fu) [reply src/mkl_pardiso.jl]: Sure, sorry did not look for all occurences.

File: src/mkl_pardiso.jl (hunk 4 above)
Comment (j-fu) [reply src/mkl_pardiso.jl]: true

File: README.md (hunk 1 above)
Comment (KristofferC) [reply README.md:34]: 👍

File: src/Pardiso.jl (hunk 3 above)
Comment (j-fu) [reply src/Pardiso.jl:139]: Basically, this test triggers an error when `using Pardiso`, after a second `using Pardiso` one nevertheless one can instantiate `MKLPardisoSolver()`. So yes, seems to be a good idea.

**Comment by j-fu** [comment]
IMHO ready to merge.

**Comment by j-fu** [comment]
Would first merge this and then retry the registry attempt at e62b47a

**Comment by KristofferC** [comment]
👍, https://github.com/JuliaRegistries/General/pull/101593

**Pull request merged**